### PR TITLE
DTSPO-15347 add azurerm_servicebus_subscription_rule resource

### PR DESCRIPTION
**Pull request by cpareek**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/DTSPO-15347

### Change description ###
added azurerm_servicebus_subscription_rule resource to this module so that the consuming repo can deploy that resource.
Tested it on civil-service repo - 
Example PR - https://github.com/hmcts/civil-service/pull/3533/files
Pipeline - https://build.hmcts.net/blue/organizations/jenkins/HMCTS_a_to_c%2Fcivil-service/detail/PR-3533/4/pipeline/202

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines

### CODE REVIEW

**Review by endakelly — APPROVED** [review]

**Review by timja — CHANGES_REQUESTED** [review]
the README should be updated with examples, you could also swap over to `terraform-docs` at the same time like in the template

**Comment by cpareek** [comment]
> the README should be updated with examples, you could also swap over to `terraform-docs` at the same time like in the template

done - had loads of problem with `terraform_docs` although I had it confiture it in the past -
this has solved it - https://github.com/yaml/pyyaml/issues/291#issuecomment-583765937

File: README.md [hunk 1]
```diff
@@ -19,26 +19,38 @@ module "servicebus-subscription" {
...
 
-### Configuration
-
-The following parameters are required by this module
-
-- `name` the name of the ServiceBus namespace.
-- `resource_group_name` the name of the resource group in which to create the ServiceBus namespace.
-- `namespace_name` the name of the service bus namespace in which the subscription must be created.
-- `topic_name` the name of the service bus topic in which the subscription must be created.
+The following example shows how to give read access to a user assigned managed identity for the subscription:
```
Comment (timja) [inline README.md:24]: I can't see this? looks like a copy paste error from line 42

Comment (cpareek) [reply README.md]: updated it now

File: README.md [hunk 2]
```diff
@@ -19,26 +19,38 @@ module "servicebus-subscription" {
 }
 ```
 
-## Variables
+## azurerm_servicebus_subscription_rule
 
-### Configuration
+The following example shows how to add azurerm_servicebus_subscription_rule:
```
Comment (timja) [inline README.md:24]:
```suggestion
The following example shows how to add a servicebus subscription rule:
```

**Review by timja — APPROVED** [review]

**Pull request merged**